src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -810,6 +828,10 @@ public void delete(long participationId, boolean deleteBuildPlan, boolean delete
             }
             // delete local repository cache
             gitService.deleteLocalRepository(repositoryUri);
+
+            if (profileService.isLocalVcsActive()) {
+                participationVCSAccessTokenRepository.deleteByParticipation_id(participationId);
+            }
```

Comment:
I'm not sure. Shouldn't the token always be deleted if a token exists. I think about config changes. The DB should always be cleaned up.